nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-core/src/main/java/org/apache/nifi/controller/repository/StandardProcessSession.java

```diff
@@ -2201,7 +2202,10 @@ public int read(final byte[] b, final int off, final int len) throws IOException
 
             @Override
             public void close() throws IOException {
-                StandardProcessSession.this.bytesRead += countingStream.getBytesRead();
+                if (!closed) {
+                    StandardProcessSession.this.bytesRead += countingStream.getBytesRead();
+                    closed = true;
+                }
```

Comment:
Agreed, it is not thread-safe. But ProcessSession is not intended to be thread-safe :)